src/util/util.c

```diff
@@ -1073,3 +1073,47 @@ int sss_rand(void)
     }
     return rand();
 }
+
+errno_t sss_canonicalize_ip_address(TALLOC_CTX *mem_ctx,
+                                    const char *address,
+                                    char **canonical_address)
+{
+    char binaddr[16];
+    const char *pcanon;
+
+    if (inet_pton(AF_INET6, address, binaddr) == 1) {
+        *canonical_address = talloc_size(mem_ctx, INET6_ADDRSTRLEN);
+        if (*canonical_address == NULL) {
+            return ENOMEM;
+        }
+
+        pcanon = inet_ntop(AF_INET6, binaddr, *canonical_address,
+                           INET6_ADDRSTRLEN);
```

Comment:
Have you considered to use `getaddrinfo()`/`getnameinfo()` with flag `NI_NUMERICHOST` which should be able to transparently handle both address types?